vassal-app/src/main/java/VASSAL/build/module/GameState.java

```diff
@@ -668,7 +683,70 @@ else if (result == JOptionPane.CANCEL_OPTION) { // "Don't Prompt Again" option.
     }
     catch (IOException e) {
       ReadErrorDialog.error(e, f);
+      return false;
     }
+
+    return true;
+  }
+
+  /**
+   * Accepts a saved file dropped onto the main window or a map, attempts to load it as a saved game.
+   * @param dtde DropTargetDropEvent from the drop() handler
+   */
+  public void dropFile(DropTargetDropEvent dtde) {
+    dtde.acceptDrop(DnDConstants.ACTION_COPY);
+    final Transferable transferable = dtde.getTransferable();
+    final DataFlavor[] flavors = transferable.getTransferDataFlavors();
+    for (final DataFlavor flavor : flavors) {
+
+      // If the drop item is a Discord file link (or a text URL), we attempt to open the connection and Load That Shit Right Off The Internet
+      if (flavor.isFlavorTextType()) {
+        try {
+          final String text = transferable.getTransferData(flavor).toString();
+
+          try {
+            final URL url = new URL(text);
+            final URLConnection uc = url.openConnection();
+            final InputStream is = new BufferedInputStream(uc.getInputStream());
```

Comment:
Use a try-with-resources here to avoid leaks:
```
try (InputStream is = uc.getInputStream(), BufferedInputStream bis = new BufferedInputStream(is)) {
```

Reply:
done